### ISSUE TITLE
Roles are not calculated for animals

### ISSUE THREAD

**Issue by hanssun07** [splintermind/Dwarf-Therapist#292]
While it would make sense that using roles on animals work, roles are simply not calculated, resulting in all 0's even after a read, but no error message.

**Comment by thistleknot** [comment]
Oh dear. Idk if that would work well. I spose it could be done if animals
were treated as there own domain. I.e..roles calculated apart from the
dwarfs.
But it wouldn't break. Just gonna mix all units values together in a
massive sorted calculation

**Comment by splintermind** [comment]
That's intended. Roles calculations are somewhat expensive and intended for fortress citizens. As thistleknot pointed out, it would also most likely throw things off for the actual citizens.

Why do you want roles for animals?

**Comment by hanssun07** [comment]
I want to be able to sort them by a combination of their stats so I can, for example, pick the top _x_ animals to save and the rest to butcher. I can do something similar with iterative filter scripts, but I can only see the ones that qualify for the filter script and it's really micromanagy to update the script every time changing only one value so I can get the right number.

I realize now that it probably would take therapist three times longer to read. Maybe add a feature to filter scripts so they can do a sort instead?

**Comment by thistleknot** [comment]
Maybe a checkbox list that allows you to select what races are calculated
for in roles (races polled from units avail)?

**Comment by splintermind** [comment]
What kind of things are you filtering on, attributes? Body size?

**Comment by hanssun07** [comment]
I'm doing a weighted sum of attributes.

**Comment by splintermind** [comment]
Yeah then the role calculations would have to be modified to run for livestock, but more importantly, group by race.

Maybe an alternative would be to have a 'scriptable' column, where the script provides the data for cell. The drawing part would be tricky since it needs to scale the values from 0-100 but it may be possible.
